amethyst_core/src/timing.rs

```diff
@@ -156,13 +151,34 @@ impl Time {
         self.time_scale = multiplier;
     }
 
-    /// Indicates a fixed update just finished.
+    /// Restarts the internal fixed update accumulator to the desired fixed update delta time.
+    ///
+    /// This should only be called by the engine.  Bad things might happen if you call this in
+    /// your game.
+    pub fn start_fixed_update(&mut self) {
+        self.fixed_time_accumulator += self.delta_seconds;
+    }
+
+    /// Checks to see if we should perform another fixed update iteration, and if so, returns true
+    /// and reduces the accumulator.
     ///
     /// This should only be called by the engine.  Bad things might happen if you call this in
     /// your game.
-    pub fn finish_fixed_update(&mut self) {
-        self.last_fixed_update += self.fixed_time
+    pub fn step_fixed_update(&mut self) -> bool {
+        match self.fixed_time_accumulator >= self.fixed_seconds {
+            true => {
+                self.fixed_time_accumulator -= self.fixed_seconds;
+                true
+            }
+            false => false,
+        }
     }
+
+    /// This function currently doesn't do anything, but is called at the end of a fixed time update series.
+    ///
+    /// This should only be called by the engine.  Bad things might happen if you call this in
+    /// your game.
+    pub fn finish_fixed_update(&mut self) {}
```

Comment:
Given it's internal only, what's the motivation for adding it now?

EDIT: I see this existed before. So, does it make sense to keep?

Reply:
I've added functionality to this function now, where it updates the current interpolation alpha factor for systems to be able to fetch.